File: communication-channels-frontend/src/app/model/Post.ts
import { User } from './User';
import { Channel } from './Channel';
import { Like } from './Like';
import { Comment } from './Comment';
import { Attachment } from './Attachment';
import { Searchable } from './Searchable';

export class Post implements Searchable {
  public id: number;
  public title: string;
  public body: string;
  public dateCreated: Date;
  public user: User;
  public channel: Channel;
  public likes: Like[] = [];
  public comments: Comment[] = [];
  public attachments: Attachment[] = [];
  public filesToUpload;
  public edited: boolean;

  getText(): string {
    return `${this.title}, post`;
  }

  type(): Post {
    return this;
  }

  toString(): string {
    return `${this.title}, post`;
  }

  getId(): number {
    return this.id;
  }
}

File: communication-channels-backend/src/main/java/com/remare/communicationchannels/entity/Post.java
package com.remare.communicationchannels.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "post")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Post {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id")
  private Long id;

  @Column(name = "title")
  private String title;

  @Column(name = "body")
  private String body;

  @Column(name = "date_created")
  @CreatedDate
  private Date dateCreated;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  private User user;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "channel_id")
  private Channel channel;

  @OneToMany(cascade = CascadeType.ALL)
  @JoinColumn(name = "post_id")
  private List<Like> likes;

  @OneToMany(fetch = FetchType.LAZY)
  @JoinColumn(name = "post_id")
  private List<Comment> comments;

  @OneToMany(fetch = FetchType.LAZY)
  @JoinColumn(name = "post_id")
  private List<Attachment> attachments;

  @Column(name = "edited")
  private boolean edited;
}
